File: src/dataset/typescript/twelve-days_Exercism-typescript-shybyte/gemini/GeminiFlash_twelve-days_v1.ts
const GIFTS = [
    'twelve Drummers Drumming',
    'eleven Pipers Piping',
    'ten Lords-a-Leaping',
    'nine Ladies Dancing',
    'eight Maids-a-Milking',
    'seven Swans-a-Swimming',
    'six Geese-a-Laying',
    'five Gold Rings',
    'four Calling Birds',
    'three French Hens',
    'two Turtle Doves',
    'a Partridge in a Pear Tree',
];

const ORDINAL_WORDS = [
    'first', 'second', 'third', 'fourth', 'fifth',
    'sixth', 'seventh', 'eighth', 'ninth', 'tenth', 'eleventh', 'twelfth'
];

class TwelveDays {
    static recite(first: number, last: number = first): string {
        let result = '';
        for (let i = first; i <= last; i++) {
            result += TwelveDays.oneVerse(i);
        }
        return result;
    }

    private static oneVerse(n: number): string {
        let giftsText = '';
        if (n === 1) {
            giftsText = GIFTS[11];
        } else {
            for (let i = n - 1; i > 0; i--) {
                giftsText = ', ' + GIFTS[12 - i] + giftsText;
            }
            giftsText = GIFTS[12 - n] + giftsText;
            giftsText = giftsText.replace(/^/, ', and ');
            giftsText = giftsText.substring(6);
        }

        return `On the ${ORDINAL_WORDS[n - 1]} day of Christmas my true love gave to me, ${giftsText}.\n`;
    }
}

export default TwelveDays;

File: src/dataset/go/twelve_days_Exercism_go_rootulp/gemini/GeminiFlash_twelve_days_v1.go
package twelve

import (
	"fmt"
	"strings"
)

var (
	ordinals = []string{
		"first", "second", "third", "fourth", "fifth", "sixth",
		"seventh", "eighth", "ninth", "tenth", "eleventh", "twelfth",
	}

	gifts = []string{
		"a Partridge in a Pear Tree",
		"two Turtle Doves",
		"three French Hens",
		"four Calling Birds",
		"five Gold Rings",
		"six Geese-a-Laying",
		"seven Swans-a-Swimming",
		"eight Maids-a-Milking",
		"nine Ladies Dancing",
		"ten Lords-a-Leaping",
		"eleven Pipers Piping",
		"twelve Drummers Drumming",
	}
)

func Verse(day int) string {
	return fmt.Sprintf("On the %s day of Christmas my true love gave to me: %s.", ordinals[day-1], clause(day))
}

func clause(day int) string {
	clauses := make([]string, day)
	for i := day - 1; i >= 0; i-- {
		if day > 1 && i == 0 {
			clauses[day-1-i] = "and " + gifts[i]
		} else {
			clauses[day-1-i] = gifts[i]
		}
	}
	return strings.Join(clauses, ", ")
}

func Song() string {
	verses := make([]string, 12)
	for day := 1; day <= 12; day++ {
		verses[day-1] = Verse(day)
	}
	return strings.Join(verses, "\n")
}